File: maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = INT_MIN;
        for(int i =0; i<nums.size();i++){
            int sum = 0;
            for(int j = i; j<nums.size();j++){
                sum += nums[j];
                ans = max(sum,ans);
            }
        }
        return ans;
    }
};

File: maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int ms = Integer.MIN_VALUE;
        int cs = 0;

        for(int i =0; i<nums.length; i++){
            cs = cs + nums[i];
            ms = Math.max(cs, ms);
            if(cs < 0){
                cs = 0;
            }
        }
        return ms;
    }
}